optimum/neuron/modeling_base.py

```diff
@@ -0,0 +1,440 @@
+        AutoConfig.register(self.model_type, AutoConfig)
+        if hasattr(self.auto_model_class, "register"):
+            self.auto_model_class.register(AutoConfig, self.__class__)
+
+    def _neuron_config_init(self, config: "PretrainedConfig"):
+        """
+        Build a Neuron config with an instance of the `PretrainedConfig` and the task.
+        """
+        # Fetch mandatory shapes from config
+        compile_shapes = {
+            key.replace("neuron_", ""): value
+            for (key, value) in config.to_diff_dict().items()
+            if key.startswith("neuron_")
+        }
+
+        # Neuron config constructuor
+        task = NeuronModel._auto_model_to_task(self.auto_model_class)
+        neuron_config_constructor = TasksManager.get_exporter_config_constructor(
+            model_type=config.model_type, exporter="neuron", task=task
+        )
+
+        # Build neuron config
+        return neuron_config_constructor(
+            config, dynamic_batch_size=getattr(config, "dynamic_batch_size", False), **compile_shapes
+        )
+
+    @classmethod
+    def get_input_static_shapes(cls, neuron_config):
+        """
+        Get a dictionary of inputs with their valid static shapes.
+        """
+        axes = neuron_config._axes
+        input_static_shapes = {
+            name: value.shape
+            for name, value in neuron_config.generate_dummy_inputs(return_tuple=False, **axes).items()
+        }
+        return input_static_shapes
+
+    def _validate_static_shape(self, input_shapes, target_shapes):
+        """
+        Check if a input needs to be padded.
+        """
+        if self.neuron_config.dynamic_batch_size is True:
+            batch_size_check = input_shapes[0] % target_shapes[0] == 0
+            other_check = input_shapes[1:] == target_shapes[1:] if len(input_shapes) > 1 else True
+            return batch_size_check and other_check
+        else:
+            return input_shapes == target_shapes
+
+    def _raise_if_invalid_padding(self, input_name, input_tensor, target_shapes, to_pad, dim):
+        if to_pad < 0:
+            extra = ", unless you set `dynamic_batch_size=True` during the compilation" if dim == 0 else ""
+            raise ValueError(
+                f"Unable to pad {input_name} with shape: {input_tensor.shape} on dimension {dim} as input shapes must be inferior"
+                f" than the static shapes used for compilation: {target_shapes}{extra}."
+            )
+
+    def _pad_to_compiled_shape(self, inputs):
+        """
+        Pad input tensors if they are not in valid shape.
+        """
+        for input_name, input_tensor in inputs.items():
+            target_shapes = self.input_static_shapes[input_name]
+            padding = ()
+            if self._validate_static_shape(input_tensor.shape, target_shapes) is True:
```

Comment:
Why `is True`? 
It is a function that always returns booleans, let's keep it the pythonic way.
```suggestion
            if self._validate_static_shape(input_tensor.shape, target_shapes):
```

Reply:
Ok, sounds good.